discovery_engine_core/core/src/stack/ops/common.rs

```diff
@@ -0,0 +1,91 @@
+// Copyright 2022 Xayn AG
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, version 3.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+use futures::{stream::FuturesUnordered, StreamExt};
+use tokio::task::JoinHandle;
+use xayn_discovery_engine_providers::{Article, Error, Market};
+
+use crate::engine::GenericError;
+
+async fn request_new_items(
+    markets: Vec<Market>,
```

Comment:
We are mixing the logic of requesting more pages with the one to request data from more then one market. Would be nice to keep them separate.